cartridge-cli/utils.lua

```diff
@@ -0,0 +1,450 @@
+local fio = require('fio')
+local errno = require('errno')
+local digest = require('digest')
+
+local utils = {}
+
+-- * -------------------- Tables --------------------
+
+function utils.array_contains(array, value)
+    if not array then
+        return false
+    end
+
+    for _, v in ipairs(array) do
+        if v == value then
+            return true
+        end
+    end
+
+    return false
+end
+
+function utils.array_index_of(array, value)
+    for i, v in ipairs(array) do
+        if v == value then
+            return i
+        end
+    end
+end
+
+function utils.dict_keys(dict)
+    local keys = {}
+
+    for key, _ in pairs(dict) do
+        table.insert(keys, key)
+    end
+    return keys
+end
+
+
+function utils.array_slice(array, from, to)
+    local result = {}
+
+    if from == nil then
+        from = 0
+    end
+
+    if to == nil then
+        to = #array
+    end
+
+    for i = from,to do
+        table.insert(result, array[i])
+    end
+
+    return result
+end
+
+function utils.merge_lists(...)
+    local res = {}
+    for i = 1, select('#', ...) do
+        local t = select(i, ...)
+        for _, v in ipairs(t) do
+            res[#res + 1] = v
+        end
+    end
+    return res
+end
+
+-- * ---------------------- Bytes ==---------------------
+
+function utils.align(addr, bytes)
+    return bit.band(addr + (bytes - 1), -bytes)
+end
+
+-- Pad the buffer with zeros so that its size is a multiple of 8 bytes
+function utils.buf_pad_to_8_byte_boundary(buf)
+    return buf .. string.rep('\0', utils.align(#buf, 8) - #buf)
+end
+
+-- * ---------------------- Strings ----------------------
+
+function utils.remove_leading_dot(filename)
+    if string.startswith(filename, '.') then
+        return string.sub(filename, 2)
+    end
+
+    return filename
+end
+
+function utils.random_string()
+    return digest.urandom(8):hex()
+end
+
+function utils.remove_leading_spaces(s, spaces_num)
+    spaces_num = spaces_num or 8
+    local REMOVE_PATTERN = string.format('^%s', string.rep(' ', spaces_num))
+
+    local res_lines = {}
+    for _, line in ipairs(s:split('\n')) do
+        local res_line = line:gsub(REMOVE_PATTERN, '')
+        table.insert(res_lines, res_line)
+    end
+
+    return table.concat(res_lines, '\n'):strip()
+end
+
+-- expand() allows to render a text template, expanding ${statement}
+-- into the calculated value of that statement.
+-- Roughly based on http://lua-users.org/wiki/TextTemplate
+--
+-- First argument is the template string, then arbitrary number of
+-- tables with mappings of variable=value
+function utils.expand(template, ...)
+    assert(type(template)=='string', 'expecting string')
+    local searchlist = {...}
+    local estring,evar
```

Comment:
`estring,evar` -> `estring, evar`

Reply:
The same as below - in a separated patch.